packages/react-utilities/src/compose/getSlots.ts

```diff
@@ -1,8 +1,87 @@
 import * as React from 'react';
 
-import { getNativeElementProps, omit } from '../utils/index';
-import { GenericDictionary } from './types';
+import { ComponentState, ShorthandRenderFunction, SlotPropsRecord } from './types';
 import { nullRender } from './nullRender';
+import { getNativeElementProps, omit } from '../utils/index';
+
+function getSlot(
+  component: React.ElementType | undefined,
+  alternative: keyof JSX.IntrinsicElements | undefined,
```

Comment:
```suggestion
  defaultComponent: React.ElementType | undefined,
  userComponent: keyof JSX.IntrinsicElements | undefined,
```

_just a proposal for naming, feel free to use something else_

As we don't need to have `defaultValue` as it fallbacks to `div` I propose to have better naming for other arguments

- first is actually a default value i.e. `components.NAME`
- while a second comes from `props` and is provided by user